src/app/labeling-suite/helper/manager/record.ts

```diff
@@ -0,0 +1,266 @@
+import { jsonCopy } from "src/app/util/helper-functions";
+import { DoBeforeDestroy } from "src/app/util/interfaces";
+import { LabelingSuiteManager, UpdateType } from "./manager";
+import { LabelingSuiteRlaPreparator } from "./recordRla";
+import { GOLD_STAR_USER_ID } from "./user";
+
+type RecordData = {
+    deleted: boolean;
+    token: any;
+    baseRecord: any;
+}
+
+export class LabelingSuiteRecordManager implements DoBeforeDestroy {
+    private baseManager: LabelingSuiteManager;
+    private recordApolloService: RecordApolloService;
+    private projectId: string;
+    public recordData: RecordData;
+    public rlaPreparator: LabelingSuiteRlaPreparator;
+
+    private activeRecordId: string;
+    private recordRequests = {
+        record: null as Subscription,
+        token: null as Subscription,
+        rla: null as Subscription,
+        combineLatest: null as Subscription,
+        rlaQuery: null as any,
+    };
+
+
+    constructor(projectId: string, recordApolloService: RecordApolloService, baseManager: LabelingSuiteManager) {
+        this.projectId = projectId;
+        this.recordApolloService = recordApolloService;
+        this.baseManager = baseManager;
+        this.rlaPreparator = new LabelingSuiteRlaPreparator(baseManager);
+        this.initRecordData();
+
+
+        NotificationService.subscribeToNotification(this, {
+            projectId: this.projectId,
+            whitelist: this.getWebsocketWhitelist(),
+            func: this.handleWebsocketNotification
+        });
+    }
+
+    public initRecordData() {
+        this.recordData = {
+            deleted: false,
+            baseRecord: null,
+            token: null,
+        };
+        this.rlaPreparator.setRlas(null, null);
+        this.baseManager.runUpdateListeners(UpdateType.RECORD);
+    }
+
+    doBeforeDestroy(): void {
+        for (const key in this.recordRequests) {
+            if (key == 'rlaQuery') continue;
+            if (this.recordRequests[key]) this.recordRequests[key].unsubscribe();
+        }
+        NotificationService.unsubscribeFromNotification(this, this.projectId);
+    }
+
+    private getWebsocketWhitelist(): string[] {
+        let toReturn = ['payload_finished', 'weak_supervision_finished'];
+        toReturn.push(...['record_deleted', 'rla_created', 'rla_deleted']);
+        return toReturn;
+    }
+
+    private handleWebsocketNotification(msgParts: string[]) {
+        if (msgParts[1] == 'record_deleted') {
+            if (msgParts[2] == this.activeRecordId) {
+                console.log("record deleted");
+                this.setDeletedState();
+            }
+        } else if (msgParts[1] == 'rla_created' || msgParts[1] == 'rla_deleted') {
+            if (this.recordRequests.rlaQuery) this.recordRequests.rlaQuery.refetch();
+        }
+        else {
+            console.log("unknown message in labeling suite record manager" + msgParts);
+        }
+
+    }
```

Comment:
`payload_finished` and `weak_supervision_finished` are part of the websocket whitelist, but are not in the handle notification method. -> leads to console log

Reply:
good catch -> included in rla query refetch